gtk4/src/expression.rs

```diff
@@ -182,6 +182,56 @@ impl glib::value::ToValueOptional for Expression {
     }
 }
 
+/// Trait containing convenience methods in creating [`gtk::PropertyExpression`] that
+/// looks up a property of a [`glib::Object`].
+pub trait PropExpr {
+    /// Create an expression looking up an object's property
+    fn property_expression(&self, property_name: &str) -> Expression;
+
+    /// Create an expression looking up an object's property with a weak reference.
+    fn weak_property_expression(&self, property_name: &str) -> Expression;
+}
+
+impl<T: IsA<glib::Object>> PropExpr for T {
+    fn property_expression(&self, property_name: &str) -> Expression {
+        let obj_expr = crate::ConstantExpression::new(self).upcast();
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name).upcast()
+    }
+
+    fn weak_property_expression(&self, property_name: &str) -> Expression {
+        let obj_expr = crate::ObjectExpression::new(self).upcast();
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name).upcast()
+    }
+}
+
+/// Trait containing convenience methods to create chains of [`gtk::PropertyExpression`]
+/// or [`gtk::ClosureExpression`].
+pub trait ChainExpr {
+    /// Create a [`gtk::PropertyExpression`] that looks up for `property_name`
+    /// with self as parameter
+    fn property_expression(&self, property_name: &str) -> Expression;
+
+    /// Create a [`gtk::ClosureExpression`] with self as a parameter
+    fn closure_expression<F, T>(self, f: F) -> Expression
+    where
+        F: Fn(&[glib::Value]) -> T + 'static,
+        T: glib::value::ValueType;
+}
+
+impl ChainExpr for Expression {
```

Comment:
```suggestion
impl<T: AsRef<Expression>> ChainExpr for T {
```

Maybe? That way it can be directly called on `PropertyExpression` / `ClosureExpression` etc? Or is that not needed because they're both `Deref<Target=Expression>`?

Reply:
I think this won't work on ClosureExpression as its parameter needs an owned Expression, unless cloning it

Reply:
They're just refcounted so that's not necessarily a problem

Reply:
Okay